pandas/core/generic.py

```diff
@@ -2434,8 +2444,22 @@ def to_msgpack(self, path_or_buf=None, encoding='utf-8', **kwargs):
         return packers.to_msgpack(path_or_buf, self, encoding=encoding,
                                   **kwargs)
 
-    def to_sql(self, name, con, schema=None, if_exists='fail', index=True,
-               index_label=None, chunksize=None, dtype=None, method=None):
+    def to_sql(self,
+               name: str,
+               con: Union['sqlalchemy.engine.Engine', 'sqlite3.Connection'],
+               schema: str = None,
+               if_exists: str = 'fail',
+               index: _bool = True,
+               index_label: Union[str, List[str]] = None,
+               chunksize: int = None,
+               dtype: Dtype = None,
+               method: Union[str,
+                             Callable[['SQLTable',
+                                       Union['sqlalchemy.engine.Engine',
+                                             'sqlite3.Connection'],
+                                       List[str], Iterator[List]], None]
+                             ] = None
```

Comment:
This isn't very nice, unfortunately...

Reply:
For such a case, would add an alias for it just before the function definition, to avoid this nesting